src/main/java/org/datadog/jenkins/plugins/datadog/DatadogClient.java

```diff
@@ -21,6 +21,21 @@
      */
     public boolean sendEvent(JSONObject payload);
 
+    /**
+     * Increment a counter for the given metrics.
+     * NOTE: To submit all counters you need to execute the flushCounters method.
+     * This is to aggregate counters and submit them in batch to Datadog in order to minimize network traffic.
+     * @param name - metric name
+     * @param hostname - metric hostname
+     * @param tags - metric tags
+     */
+    public void incrementCounter(String name, String hostname, JSONArray tags);
+
+    /**
+     * Submit all your counters as rate with 10 seconds intervals.
+     */
+    public void flushCounters();
```

Comment:
Shouldn't this be a flushAll method, that would flush gauges and service checks and everything every 10s ?

Reply:
Maybe we should do the same for metrics and services checks indeed. However I will not do it in this PR.